ladybug/src/main/java/nl/nn/ibistesttool/larva/ConvertToLarvaAction.java

```diff
@@ -0,0 +1,322 @@
+/*
+   Copyright 2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.ibistesttool.larva;
+
+import nl.nn.testtool.Checkpoint;
+import nl.nn.testtool.Report;
+import nl.nn.testtool.extensions.CustomReportAction;
+import nl.nn.testtool.extensions.CustomReportActionResult;
+
+import nu.studer.java.util.OrderedProperties;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.OptionalInt;
+import java.util.stream.Stream;
+
+public class ConvertToLarvaAction implements CustomReportAction {
+
+	String dir;
+	CustomReportActionResult customReportActionResult;
+
+	@Override
+	public String getButtonText() {
+		return "Convert to Larva";
+	}
+
+	public void setReportsFolder(String dir) {
+		this.dir = dir;
+	}
+
+	@Override
+	public CustomReportActionResult handleReports(List<Report> reports) {
+		List<Report> successReports = new ArrayList<>();
+		customReportActionResult = new CustomReportActionResult();
```

Comment:
Dit is niet threadsafe
CustomReportActionResult moet in deze methode als veld gedefineerd worden.